### DISCUSSION TITLE
Widget visibility

### DISCUSSION THREAD

**Discussion by invaders-xx** [filamentphp/filament#2124] [Help]
I registered a new Widget and I want to only display this widget on specific page and not on the main dashboard. Is there a way to handle that ?

**Comment by invaders-xx** [comment]
I tried to play with canView() in the widget and test the request()->routeIs('route-to-dashboard'), it works but it's quite ugly, isn't it ?

**Reply by danharrin** [reply]
Dont put the widgets class in the Widgets namespace, but it in the Pages/PageName/Widgets namespace. Then it wont be autodiscovered for the dashboard

**Reply by alboxe** [reply]
This does the trick. I installed the Fullcalendar plugin en the Calendar widget repressed my other widgets on the admin homepage.
After putting the widget in another folder/namespace it went well.

**Comment by invaders-xx** [comment]
In fact the Widget is not located in the app/Filament structure but in a module (nwidart package) and my concern was to make this widget only visible within the dashboard page of the module and not in the main dashboard page. As I have to register the widget, it automatically appears in the main dashboard if I don't restrict with canView method. but maybe there is another way

**Reply by danharrin** [reply]
If you register a widget using `Livewire::component('livewire.component.name', LivewireComponentClass::class)` instead of `Filament::registerWidgets()`, it will have the same affect.

Alternatively, if your widget is associated with a particular resource, you could register it in the resource's `getWidgets()` method instead of `Filament::registerWidgets()`.

**Reply by invaders-xx** [reply]
Thanks @danharrin. I registered widgets in module's provider instead of Filament and then I can display those widget in my module's relation dashboard pages. Many thanks once again for your precious and effective help.
